client/me/purchases/manage-purchase/notices.tsx

```diff
@@ -103,6 +103,15 @@ class PurchaseNotice extends Component<
 		if ( isMonthly( purchase.productSlug ) ) {
 			const daysToExpiry = moment( expiry.diff( moment() ) ).format( 'D' );
 
+			if ( isTemporarySitePurchase( purchase ) ) {
+				return translate( '%(purchaseName)s will expire and be removed %(expiry)s days. ', {
```

Comment:
I'm not certain myself, but testing locally, it seems it... does add the 'in' for cases where the expiry is framed in terms of days? This doesn't make any sense to me, TBH, but manually changing the expiry date for my purchase still gives me the correct message. Below is a screenshot after I changed it to a date 14 days in the future. But since this is such unusual behavior (IMO), I encourage anybody testing to double-check me.

<img width="1059" alt="image" src="https://github.com/Automattic/wp-calypso/assets/670067/f5068bf3-cd13-45cb-92cb-0ec74a59e6b8">

Reply:
That's... very strange! 🙃  I tried this in a playground and this code definitely does not add 'in':

```js
const expiry = moment( '2023-10-05 00:00Z' );
console.log( moment( expiry.diff( moment() ) ).format( 'D' ) );
```

In fact, it doesn't return a number of days at all. Instead it interprets the difference between the days as a day and then displays that day's number; in this case, `19`.

To get the number of days until something happens, you have to do this:

```js
const expiry = moment( '2023-10-05 00:00Z' );
console.log( expiry.diff( moment(), 'days' ) );
```

Which outputs `50`, the number of days between now and 5 October.

Given that, I don't see how this has ever worked correctly or how it's working now. 

You'll notice that the other strings in this file, like [this one](https://github.com/Automattic/wp-calypso/blob/d17dd34d7c22a17991e0ad2d7bdca773a6d50b5d/client/me/purchases/manage-purchase/notices.tsx#L139), use `momentObject.fromNow()`. If I do this:

```
const expiry = moment( '2023-10-05 00:00Z' );
console.log( expiry.fromNow() );
```

I get `in 2 months`, which is correct and looks like what you're aiming for here, but `daysToExpiry` does not look like it has the right value.

That said, I can't explain how you're getting the screenshots you've provided. Given the above information, what do you think is happening?

Reply:
Just dropping by to confirm, I also see the correct text 

![Screenshot 2023-08-16 at 8 55 39 AM](https://github.com/Automattic/wp-calypso/assets/12895386/1893deac-579a-41c0-ba5a-695316e00ad0)

Reply:
I see something similar, but in my case, the text is `in a year`, which is clearly not the string we're looking at here, which includes a hard-coded `days` word.

<img width="1074" alt="Screenshot 2023-08-16 at 10 40 23 AM" src="https://github.com/Automattic/wp-calypso/assets/2036909/cff1509a-37ef-4218-94f2-b06338924865">

I believe what we're both seeing is the text from [this line](https://github.com/Automattic/wp-calypso/blob/2b9d6aa97423edc312f9790014c9e6e02c2efdde/client/me/purchases/manage-purchase/notices.tsx#L127), which uses `expiry.fromNow()` as I mentioned above.

The reason is that this code is inside the `isMonthly()` check and the testing instructions don't mention that the product must be monthly to test these changes. If I modify the code to use this version, here's what I see:

<img width="630" alt="Screenshot 2023-08-16 at 10 44 53 AM" src="https://github.com/Automattic/wp-calypso/assets/2036909/7c9412d5-f4bc-400d-9ef0-c1be57c23704">

Note that even apart from the strange wording, the "1" is incorrect; this subscription expires in about 365 days.

This is definitely wrong and the one below it (in trunk) is also broken.